### PR TITLE
add workflow support

### PR DESCRIPTION
**Pull request by w10u**

### CODE REVIEW

File: pages/endpoint/workflow.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import Heurist from 'heurist'
+
+const heurist = new Heurist({
+  apiKey: process.env['HEURIST_API_KEY'], // This is the default and can be omitted
+})
+
+async function main() {
+  const upscalerTask = new UpscalerTask({
+    consumer_id: 'example-id',
+    workflow_id: '1',
```
Comment (wjw12) [inline pages/endpoint/workflow.mdx]: Is there 1:1 mapping between workflow ID and Task? Like upscaling workflow is ID1, animation workflow is ID2, and so on, defined in the server/miner side? If so, we can make the workflow ID a static variable of the Task class

File: packages/workflow/index.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    const result = await response.json()
+    return result.task_id
+  }
+
+  async executeWorkflowAndWaitForResult(
+    task: WorkflowTask,
+    timeout: number = 300000,
+    interval: number = 10000
```
Comment (wjw12) [inline packages/workflow/index.ts:137]: Let's set a const MIN_INTERVAL = 1000 ms to restrict the interval being too short and may effectively create a DDOS

File: pages/endpoint/workflow.mdx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+- Type: `UpscalerTask`
+
+<OptionTable
+  language="en"
+  options={[
+    [
+      'consumer_id',
+      'string',
+      true,
+      'The ID of the consumer initiating the task.',
```
Comment (wjw12) [inline pages/endpoint/workflow.mdx]: What's the concept and requirement of consumer_id? can add more description?

File: pages/endpoint/workflow.mdx (hunk 1 above)
Comment (ConnorLi96) [reply pages/endpoint/workflow.mdx]: No such mapping now, the original idea is miner side will distinguish different task type and send it to correct docker service. User end actually no need to send workflow_id, just use task type.

Comment (w10u) [reply pages/endpoint/workflow.mdx]: I thought the workflow_id can distinguish different versions of, say, UpscalerTask (like 1 being simple upscaler, 2 being more complex upscaler to 4k or something)? Is there such a use case? If not then we just need task type

Comment (w10u) [reply pages/endpoint/workflow.mdx]: wil remove workflow_id as discussed on TG

File: pages/endpoint/workflow.mdx (hunk 3 above)
Comment (w10u) [reply pages/endpoint/workflow.mdx]: updated

**Pull request merged**